src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -141,6 +148,11 @@ public string GetShortFolderName()
             return GetShortFolderName(DefaultFrameworkNameProvider.Instance);
         }
 
+        /// <summary>
+        /// True if this NuGetFramework is not valid and therefore not functional.
+        /// </summary>
+        public bool IsInvalid { get; private set; }
```

Comment:
I don't like negative nouns, because when it's false, it's a double negative. So, if we keep this, I think inverting it to `IsValid` would be better.

However, it sounds like it should have the same value as `IsSupported`. Why not just extend that property, rather than adding another?

Reply:
Make sure you explicitly set the inverted noun to "true". Right now, the thing I don't like is happening, which is the implicit value is set by default(bool).

Reply:
yes, it's possible to set a boolean's default to true by either using the `public bool IsInvalid {get; private set; } = true;` syntax, or simply setting it to true in the constructor.  Personally, I don't remember hearing guidelines about always using booleans with default false values, whereas I had heard of naming conventions to avoid false values being a double-negative.

Reply:
Good suggestion. Looked as IsUnsupported, which is a bit wacky, but i like extending that to also look at a private IsInvalid.
Fixed in next commit.